src/pages/vaults/VaultDepositWithdrawForm.tsx

```diff
@@ -0,0 +1,637 @@
+      if (freeCollateralUpdated == null || freeCollateralUpdated < 0) {
+        allErrors.push({
+          type: AlertType.Error,
+          key: 'deposit-high',
+          long: stringGetter({ key: STRING_KEYS.DEPOSIT_TOO_HIGH }),
+          short: 'Modify amount',
+        });
+      }
+    } else {
+      if (userBalanceUpdated == null || userBalanceUpdated < 0) {
+        allErrors.push({
+          type: AlertType.Error,
+          key: 'withdraw-high',
+          long: stringGetter({ key: STRING_KEYS.WITHDRAW_TOO_HIGH }),
+          short: 'Modify amount',
+        });
+      }
+      if (slippagePercent >= SLIPPAGE_PERCENT_WARN) {
+        allErrors.push({
+          type: AlertType.Warning,
+          key: 'slippage-high',
+          long: (
+            <span>
+              {stringGetter({
+                key: STRING_KEYS.SLIPPAGE_WARNING,
+                params: {
+                  AMOUNT: <$InlineOutput value={slippagePercent} type={OutputType.Percent} />,
+                  LINK: (
+                    <Link href={vaultsLearnMore} withIcon isInline>
+                      {stringGetter({ key: STRING_KEYS.VAULT_FAQS })}
+                    </Link>
+                  ),
+                },
+              })}
+            </span>
+          ),
+        });
+        if (slippagePercent >= SLIPPAGE_PERCENT_ACK && !slippageAck && currentForm === 'confirm') {
+          allErrors.push({
+            type: AlertType.Error,
+            key: 'slippage-acked',
+            short: stringGetter({ key: STRING_KEYS.ACKNOWLEDGE_HIGH_SLIPPAGE }),
+          });
+        }
+      }
+    }
+
+    return allErrors;
+  }, [
+    amount,
+    canViewAccount,
+    currentForm,
+    freeCollateralUpdated,
+    isAccountViewOnly,
+    selectedType,
+    slippageAck,
+    stringGetter,
+    userBalanceUpdated,
+    vaultsLearnMore,
+  ]);
+
+  const onSubmitInputForm = useCallback(() => {
+    setCurrentForm('confirm');
+  }, []);
+
+  const onSubmitConfirmForm = useCallback(() => {
+    // TODO tell abacus and respond
+    onSuccess?.();
+  }, [onSuccess]);
+
+  const onClickMax = useCallback(() => {
+    if (selectedType === 'deposit') {
+      setAmountState(`${freeCollateral ?? ''}`);
+    } else {
+      setAmountState(`${userBalance ?? ''}`);
+    }
+  }, [freeCollateral, selectedType, userBalance]);
+
+  const freeCollateralDiff = renderDiffOutput({
+    type: OutputType.Fiat,
+    value: freeCollateral?.current,
+    newValue: freeCollateralUpdated,
+    withDiff:
+      MustBigNumber(amount).gt(0) &&
+      freeCollateralUpdated != null &&
+      freeCollateral?.current !== freeCollateralUpdated,
+  });
+  const vaultDiff = renderDiffOutput({
+    type: OutputType.Fiat,
+    value: userBalance,
+    newValue: userBalanceUpdated,
+    withDiff:
+      MustBigNumber(amount).gt(0) &&
+      userBalanceUpdated != null &&
+      userBalanceUpdated !== userBalance,
+  });
+  const marginUsageDiff = renderDiffOutput({
```

Comment:
[non-blocking] this is more of a style preference, but unless these are going to be reused i'd probably prefer to see them inline to remove a layer of indirection + colocate logic. but definitely not blocking

Reply:
All of these are used in more than one place except marginUsageDiff, which I kept next to the others for consistency.